tests/test.ts

```diff
@@ -20,6 +21,107 @@ const test_cases: {
   '々': 'cjk-letter',
 };
 
+// BMP
+const cjkRadicalsSupplement = '⻁⺪⻏';
+const kangxiRadicals = '⽒⽓⽔⽕';
+const ideographicDescriptionCharacters = '⿰⿱⿲⿳';
+const cjkSymbolsPunctuation = '「」〒〓〱〻〧〨';
+const hiragana = 'どなにぬ';
+const katakana = 'ハヒヘホ';
+const bopomofo = 'ㄬㄭㄮ';
+const alsoBopomofo = 'ㄯ';
+const hangulCompatibilityJamo = 'ㅵㅶㅷㅸ';
+const kanbun = '㆐㆑㆒㆓';
+const bopomofoExtended = 'ㆠㆡ';
+const alsoBopomofoExtendedIntroducedInUnicode13 = 'ㆾㆿ';
+const cjkStrokes = '㇀㇁㇂㇃㇄㇅㇆㇇㇈㇉㇊㇢㇣';
+const katakanaPhoneticExtensions = 'ㇼㇽㇾㇿ';
+const enclosedCjkLettersMonths = '㈱㋋㋌㋿';
+const cjkCompatibility = '㌕㍕㍖㍗';
+const alsoCjkCompatibility = '㍧㍻㍿㏄㏾';
+const cjkUnifiedIdeographsExtensionA = '䶼䶽䶾䶿';
+const hangulJamoExtendedA = 'ꥠꥹꥺꥻ';
+const hangulSyllables = '힠힡힢힣';
+const hangulJamoExtendedB = 'ퟸퟹퟺퟻ';
+const cjkCompatibilityIdeographs = '豈兀﨔隷侮恵並﨎';
+const cjkCompatibilityForms = '﹌﹍﹎﹏';
+const smallFormVariants = '﹨﹩﹪﹫';
+const halfwidthandFullwidthForms = '！Ｄ｢￥￮';
+const alsoHalfwidthandFullwidthForms = 'ﾕￇ';
+
+// SMP
+const kanaExtendedB = '𚿵𚿶𚿷𚿸';
+const kanaSupplement = '𛀀𛀁𛃾𛃿';
+const kanaExtendedA = '𛄀𛄁𛄡𛄢';
+const smallKanaExtension = '𛅕𛄲𛅤𛅥𛅦𛅧';
+const nushu = '𛋸𛋹𛋺𛋻';
+const khitanSmallScript = '𘳒𘳓𘳔𘳕';
+const cjkUnifiedIdeographsExtensionB = '𪛜𪛝𪛞𪛟';
+const cjkUnifiedIdeographsExtensionC = '𫚄𫚅𫜸𫜹';
+const cjkUnifiedIdeographsExtensionD = '𫝀𫝁𫝂𫝃';
+const cjkUnifiedIdeographsExtensionE = '𬷰𬷱𬺠𬺡';
+const cjkUnifiedIdeographsExtensionF = '𬻿𮨥𮨦𮯠';
+const cjkUnifiedIdeographsExtensionG = '𰀀𰀁𱍀𱍊';
+const cjkUnifiedIdeographsExtensionH = '𱍐𱍑𲎬𲎭';
+const cjkCompatibilityIdeographsSupplement = '乁你鼻𪘀';
+
+function casify(chars: string, t: TestType): TestCases {
+  return Object.fromEntries(chars.split('').map(c => [c, t]));
+}
+
+const more_test_cases: TestCases = {
+  ...casify(cjkRadicalsSupplement, 'cjk-letter'),
+  ...casify(cjkSymbolsPunctuation.slice(0, 5), 'cjk-punctuation'),
+  ...casify(cjkSymbolsPunctuation.slice(6, 8), 'cjk-letter'),
+  ...casify(kangxiRadicals, 'cjk-letter'),
+  ...casify(ideographicDescriptionCharacters, 'cjk-punctuation'),
+  ...casify(hiragana + katakana, 'cjk-letter'),
+  ...casify(katakanaPhoneticExtensions, 'cjk-letter'),
+  ...casify(bopomofo + bopomofoExtended, 'cjk-letter'),
+  ...casify(
+    hangulCompatibilityJamo + hangulJamoExtendedA + hangulJamoExtendedB,
+    'cjk-letter',
+  ),
+  ...casify(hangulSyllables, 'cjk-letter'),
+  ...casify(cjkCompatibility, 'cjk-letter'),
+  ...casify(cjkCompatibilityIdeographs, 'cjk-letter'),
+  ...casify(
+    cjkCompatibilityForms + smallFormVariants + halfwidthandFullwidthForms,
+    'cjk-punctuation',
+  ),
+  ...casify(alsoHalfwidthandFullwidthForms, 'cjk-letter'),
+  ...casify(nushu, 'non-cjk'),
+  ...casify(khitanSmallScript, 'non-cjk'),
+};
+
+const should_be_cjk_but_currently_not_matched: TestCases = {
```

Comment:
Are you interested in addressing the not matched issue in this PR as well?

Reply:
Not in this PR (I am not good at writting Regex).

Reply:
Just FYI, we actually write unicode category instead of regex here, see https://github.com/ikatyang/cjk-regex/blob/v3.0.0/src/index.ts#L4-L25 for example.

Reply:
not matched issue resolved in [ed9373e](https://github.com/ikatyang/cjk-regex/pull/178/commits/ed9373e403e0714bb9c060813c1c9367abb873ca)